tests/test_orient_lud.py

```diff
@@ -0,0 +1,193 @@
+        tol = 9
+    mean_aligned_angular_distance(est_rots, source.rotations, degree_tol=tol)
+
+    # Check dtype pass-through
+    np.testing.assert_equal(source.dtype, est_rots.dtype)
+
+
+def test_compute_num_eigs():
+    eig_vec = np.array([2, 2, 2, 2, 2, 2, 2, 2, 1])  # max drop = 2/1 at idx=7
+    rel_drp_thresh = 5
+    eigs_inc = 2
+    num_eigs_W = 15
+
+    # Case for num_eigs_prev = 0:
+    num_eigs_prev = 0
+    num_eigs = CommonlineLUD._compute_num_eigs(
+        num_eigs_prev, eig_vec, num_eigs_W, rel_drp_thresh, eigs_inc
+    )
+    np.testing.assert_equal(num_eigs, 6)
+
+    # Cases for num_eigs_prev > 0:
+    num_eigs_prev = 7
+
+    # For len(eig_vec) = 1 , output = num_eigs_prev + eigs_inc = 7 + 2
+    ev = eig_vec[:1]
+    num_eigs = CommonlineLUD._compute_num_eigs(
+        num_eigs_prev, ev, num_eigs_W, rel_drp_thresh, eigs_inc
+    )
+    np.testing.assert_equal(num_eigs, 9)
+
+    # For len(eig_vec) = 2 , output = 6
+    ev = eig_vec[:2]
+    num_eigs = CommonlineLUD._compute_num_eigs(
+        num_eigs_prev, ev, num_eigs_W, rel_drp_thresh, eigs_inc
+    )
+    np.testing.assert_equal(num_eigs, 6)
+
+    # For len(eig_vec) > 2 and rel_drp > rel_drp_thresh, output = num_eigs_prev + eigs_inc = 7 + 2
+    # where rel_drp = (num_eigs_W - 1) * max_drop / (np.sum(drops) - max_drop) = 14 * 2 / (9 - 2) = 4
+    num_eigs = CommonlineLUD._compute_num_eigs(
+        num_eigs_prev, eig_vec, num_eigs_W, rel_drp_thresh, eigs_inc
+    )
+    np.testing.assert_equal(num_eigs, 9)
+
+    # For len(eig_vec) > 2 and rel_drp <= rel_drp_thresh, output = index of max drop in elements of eig_vec = 7
+    # Setting rel_drp_thresh = 3 (ie. < 4)
+    num_eigs = CommonlineLUD._compute_num_eigs(
+        num_eigs_prev, eig_vec, num_eigs_W, 3, eigs_inc
+    )
+    np.testing.assert_equal(num_eigs, 7)
+
+
+def test_compute_AX_ATy():
+    """
+    Test we get the intended result for `_compute_AX()`, where A(X) is defined as:
+
+        A(X) = [
+            X_ii^(11),
+            X_ii^(22),
+            sqrt(2)/2 * X_ii^(12) + sqrt(2)/2 * X_ii^(21)
+        ]
+
+        i = 1, 2, ..., K
+
+        where X_{ii}^{pq} denotes the (p,q)-th element in the 2x2 sub-block X_{ii}.
+
+    Also test ATy. For a 2x2 block diagonal matrix, X, if A(X) = y, AT(y) = X.
+    """
+    # We create a symmetric 2k x 2k matrix with specific 2x2 blocks along the diagonal
+    k = 3
+    X = np.zeros((2 * k, 2 * k))
+
+    # Create symmetric 2x2 blocks to go along the diagonal of X
+    X[:2, :2] = np.array([[1.0, 2.0], [2.0, 1.0]])
+    X[2:4, 2:4] = np.array([[3.0, 4.0], [4.0, 3.0]])
+    X[4:, 4:] = np.array([[5.0, 6.0], [6.0, 5.0]])
+
+    # Check the result. We should have:
+    y_gt = np.array([1, 1, 3, 3, 5, 5, np.sqrt(2) * 2, np.sqrt(2) * 4, np.sqrt(2) * 6])
+    y = CommonlineLUD._compute_AX(X)
+    np.testing.assert_allclose(y, y_gt)
+
+    # Check ATy(y) = X
+    np.testing.assert_allclose(CommonlineLUD._compute_ATy(y), X)
```

Comment:
May also be good to test the adjoint property, i.e., for any `u` and `v` (could be drawn randomly), we make sure that `(A u, v) = (u, A^T v)`.